libraries/botbuilder/src/teamsInfo.ts

```diff
@@ -414,11 +414,12 @@ export class TeamsInfo {
      * @private
      */
     private static getConnectorClient(context: TurnContext): ConnectorClient {
-        if (!context.adapter || !('createConnectorClient' in context.adapter)) {
+        const client = context.turnState?.get<ConnectorClient>(context.adapter.ConnectorClientKey);
+        if (!client) {
```

Comment:
@joshgummersall @stevengum This change ensures parity with .NET. However, the necessity for test changes make me a little wary. Any thoughts on using `context.adapter` as a fallback? Something like:

```js
const client = context.turnState?.get<ConnectorClient>(context.adapter.ConnectorClientKey) ?? 'createConnectorClient' in context.adapter ? (context.adapter as BotFrameworkAdapter).createConnectorClient(context.activity.serviceUrl) : null;
    if (!client) {
		throw new Error('This method requires a connector client.');
	}
	return client;
}
        
```

If we go this route, I'm open to suggestions for making that cleaner.

Reply:
It looks like `BotFrameworkAdapter` already stashes the `connectorClient` inside the `turnState`. Is the issue, then, just test adapters?

Reply:
I believe it's only with test adapters, yes, which is why I submitted with .NET parity. Just wanted to call it out in case I'm missing something.

Reply:
@stevengum the only thing I can think of is existing users that rely on the adapter method. What do you think? We kind of have an implicit contract with respect to what exists in turn state.

Reply:
To me this is a spot on compromise for obtaining the `connectorClient`. With @joshgummersall's comment:

> @stevengum the only thing I can think of is existing users that rely on the adapter method. What do you think? We kind of have an implicit contract with respect to what exists in turn state.

We should consider having `TeamsInfo.getConnectorClient()` defer to `context.adapter` first if `createConnectorClient` is a member. _Then_ fallback to using the `context.turnState` for the `ConnectorClient`.

Reply:
Resolved. We had some discussions in Teams and decided the best approach was to prefer to use `context.adapter` and fallback to `turnState.get()`, similar to the code I posted above, but just with the logic flipped.